x-pack/platform/packages/shared/onechat/onechat-server/tools/builtin.ts

```diff
@@ -5,13 +5,69 @@
  * 2.0.
  */
 
+import type { MaybePromise } from '@kbn/utility-types';
 import type { z, ZodObject } from '@kbn/zod';
+import type { IUiSettingsClient } from '@kbn/core-ui-settings-server';
 import type { ToolDefinition, ToolType } from '@kbn/onechat-common';
 import type { EsqlToolDefinition } from '@kbn/onechat-common/tools/types/esql';
 import type { IndexSearchToolDefinition } from '@kbn/onechat-common/tools/types/index_search';
 import type { WorkflowToolDefinition } from '@kbn/onechat-common/tools/types/workflow';
+import type { KibanaRequest } from '@kbn/core-http-server';
 import type { ToolHandlerFn } from './handler';
 
+/**
+ * Information exposed to the {@link ToolAvailabilityHandler}.
+ */
+export interface ToolAvailabilityContext {
+  request: KibanaRequest;
+  uiSettings: IUiSettingsClient;
+  spaceId: string;
+}
+
+/**
+ * Information exposed to the {@link ToolAvailabilityHandler}.
+ */
+export interface ToolAvailabilityResult {
+  /**
+   * Whether the tool is available or not.
+   */
+  status: 'available' | 'unavailable';
```

Comment:
Later we could add `disabled`, to show the tools in the UI while not letting the user use it (e.g product_doc if not installed), but let's keep it simple for now.